src/main/java/com/google/cloud/datafusion/plugin/ariba/source/AribaBatchSource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.datafusion.plugin.ariba.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.cloud.datafusion.plugin.ariba.source.config.AribaPluginConfig;
+import com.google.cloud.datafusion.plugin.ariba.source.exception.AribaException;
+import com.google.cloud.datafusion.plugin.ariba.source.util.ResourceConstants;
+import com.google.cloud.datafusion.plugin.ariba.source.util.Util;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import org.apache.hadoop.io.LongWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+/**
+ * Batch Source that reads data from an Ariba system.
+ * <p>
+ * LongWritable is the first parameter because that is the key used by Hadoop's
+ * {@code TextInputFormat}. Similarly, Text is the second parameter because that
```

Comment:
Text is not the second parameter and we have two StructuredRecords?

Reply:
Changed the doc, and Yes we have two StructureRecords. I have taken example from 'sap-plugins' and 'servicenow-plugins'